src/librustc_resolve/lifetimes.rs

```diff
@@ -1832,18 +1887,38 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                 lifetime_ref
             );
             err.span_label(lifetime_ref.span, "undeclared lifetime");
-            if !self.is_in_fn_syntax {
-                for generics in &self.missing_named_lifetime_spots {
-                    let (span, sugg) = match &generics.params {
-                        [] => (generics.span, format!("<{}>", lifetime_ref)),
-                        [param, ..] => (param.span.shrink_to_lo(), format!("{}, ", lifetime_ref)),
-                    };
-                    err.span_suggestion(
-                        span,
-                        &format!("consider introducing lifetime `{}` here", lifetime_ref),
-                        sugg,
-                        Applicability::MaybeIncorrect,
-                    );
+            for missing in &self.missing_named_lifetime_spots {
```

Comment:
Could you factor this out into a function and move it into `diagnostics.rs`.

Reply:
I ended up moving a few of these to methods in `diagnostics.rs` to avoid having too many free-standing functions that are all otherwise related to `LifetimeContext`.